xlators/mgmt/glusterd/src/glusterd.c

```diff
@@ -1621,11 +1621,39 @@ init(xlator_t *this)
         exit(1);
     }
 
-    snprintf(cmd_log_filename, PATH_MAX, "%s/cmd_history.log", logdir);
+    char *cmd_history_file = NULL;
+    ret = dict_get_str_sizen(this->options, "command-history-file",
+                             &cmd_history_file);
+    if (!cmd_history_file) {
+        snprintf(cmd_log_filename, PATH_MAX, "%s/cmd_history.log", logdir);
+    } else {
+        if (cmd_history_file[0] != '/') {
```

Comment:
Do we allow to store the file anywhere in the filesystem, even on mounted subfilesystems ? it's a potential security issue.

I think we should limit the configuration option to the name of the file. The directory should always be `logdir`. Is there any use case that requires to move that file outside that directory ?

Reply:
Ack, I do agree its an CVE issue, and previously we have fixed similar issues in handling other options.